crypto/params.c

```diff
@@ -398,7 +425,8 @@ int OSSL_PARAM_get_int32(const OSSL_PARAM *p, int32_t *val)
                 *val = (int32_t)d;
                 return 1;
             }
-            break;
+            err_out_of_range;
+            return 0;
         }
     }
     return 0;
```

Comment:
Why no error report here?

Reply:
I was concentrating on errors of the "doesn't fit the size of the destination" type.  This was the complaint that caused Matt to update the documentation and me to add some errors.  It's a subtle problem IMO because there is nothing obviously incorrect about the calls being made.

This failure (and later ones) are of the form "parameter type is incorrect".  I did start looking at all of the `return 0;` statements.  It needed a number of additional errors (easy) but they were for far more obvious mistakes.  It was more refactoring than I thought worthwhile for the limited benefit.

Adding errors for every failure certainly is possible.  It spreads quite a bit wider than just `params.c`.

Reply:
I'll be satisfied if there's an issue raised for getting the rest done. I would rather not end up with functions that sometimes do and sometimes don't add an error record when there is an error. Doing this in multiple PRs is absolutely fine. Leaving it incomplete in the long run isn't.

Reply:
I'll add extra errors, I don't see there being any rush to get this in.